cpp/src/aztec3/circuits/kernel/private/.test.cpp

```diff
@@ -70,288 +61,256 @@ using aztec3::circuits::apps::test_apps::escrow::deposit;
+    FunctionExecutionContext<aztec3::circuits::kernel::private_kernel::Composer>&,
+    std::array<NT::fr, aztec3::ARGS_LENGTH> const&)>;
+
+// Some helper constants for trees
+constexpr size_t MAX_FUNCTION_LEAVES = 2 << (aztec3::FUNCTION_TREE_HEIGHT - 1);
+const NT::fr EMPTY_FUNCTION_LEAF = FunctionLeafPreimage<NT>{}.hash(); // hash of empty/0 preimage
+const NT::fr EMPTY_CONTRACT_LEAF = NewContractData<NT>{}.hash();      // hash of empty/0 preimage
+const auto& EMPTY_FUNCTION_SIBLINGS = compute_empty_sibling_path<NT, aztec3::FUNCTION_TREE_HEIGHT>(EMPTY_FUNCTION_LEAF);
+const auto& EMPTY_CONTRACT_SIBLINGS = compute_empty_sibling_path<NT, aztec3::CONTRACT_TREE_HEIGHT>(EMPTY_CONTRACT_LEAF);
+
 } // namespace
 
 namespace aztec3::circuits::kernel::private_kernel {
 
-class private_kernel_tests : public ::testing::Test {};
-
-TEST(private_kernel_tests, test_deposit)
+/**
+ * @brief Print some debug info about a composer if in DEBUG_PRINTS mode
+ *
+ * @param composer
+ */
+void debugComposer(Composer const& composer)
 {
-    //***************************************************************************
-    // Some private circuit proof (`deposit`, in this case)
-    //***************************************************************************
-
-    const NT::address escrow_contract_address = 12345;
-    // const NT::fr escrow_contract_leaf_index = 1;
-    const NT::fr escrow_portal_contract_address = 23456;
-
-    const NT::fr msg_sender_private_key = 123456789;
-    const NT::address msg_sender =
-        NT::fr(uint256_t(0x01071e9a23e0f7edULL, 0x5d77b35d1830fa3eULL, 0xc6ba3660bb1f0c0bULL, 0x2ef9f7f09867fd6eULL));
-    const NT::address tx_origin = msg_sender;
+#ifdef DEBUG_PRINTS
+    info("computed witness: ", composer.computed_witness);
+    // info("witness: ", private_kernel_composer.witness);
+    // info("constant variables: ", private_kernel_composer.constant_variables);
+    // info("variables: ", composer.variables);
 
-    Composer deposit_composer = Composer("../barretenberg/cpp/srs_db/ignition");
-    DB db;
+    // TODO: this fails intermittently, with:
+    // bigfield multiply range check failed
+    info("failed?: ", composer.failed());
+    info("err: ", composer.err());
+    info("n: ", composer.get_num_gates());
+#else
+    (void)composer; // only used in debug mode
+#endif
+}
 
-    FunctionData<NT> function_data{
-        .function_selector = 1, // TODO: deduce this from the contract, somehow.
+/**
+ * @brief Generate a verification key for a private circuit.
+ *
+ * @details Use some dummy inputs just to get the VK for a private circuit
+ *
+ * @param is_constructor Whether this private call is a constructor call
+ * @param func The private circuit call to generate a VK for
+ * @param num_args Number of args to that private circuit call
+ * @return std::shared_ptr<NT::VK> - the generated VK
+ */
+std::shared_ptr<NT::VK> gen_func_vk(bool is_constructor, private_function const& func, size_t const num_args)
```

Comment:
We might be able to remove this function when we complete:
- https://github.com/AztecProtocol/aztec3-circuits/issues/209
and/or
- https://github.com/AztecProtocol/aztec3-circuits/issues/200

Reply:
We need it for ContractDeploymentData and for function tree leaves

Reply:
This generates the VK for a private call, not a mock kernel VK.

Reply:
Ah, thanks for explaining